### PR TITLE
Add go version from go.mod in GH Workflow

### LINKED ISSUE #101: update go setup to use go.mod
```
      uses: actions/setup-go@v4
      with:
        go-version-file: go.mod
```
sets go up based on the version is go.mod

### PR DESCRIPTION
**Pull request by husky-parul**
- Fixes #101 
Signed-off-by: Parul Singh <parsingh@redhat.com>

### CODE REVIEW

File: sm.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+
```
Comment (sthaha) [inline sm.yaml:1]: Was this added by mistake ?

**Review by sthaha — CHANGES_REQUESTED** [review]
Could you please resubmit with the unrelated file removed :) ?

**Comment by husky-parul** [comment]
@sthaha Please TAL

**Review by sthaha — APPROVED** [review]
/lgtm

**Pull request merged**